modin/engines/base/frame/partition_manager.py

```diff
@@ -280,6 +280,50 @@ def map_axis_partitions(cls, axis, partitions, map_func, keep_partitioning=False
         # the structure to the correct order.
         return result_blocks.T if not axis else result_blocks
 
+    @classmethod
+    def shuffle_simple(cls, axis, partitions, map_func, lengths):
+        """
+        Shuffle data using `lengths` via `map_func`
+
+        Parameters
+        ----------
+            axis : 0 or 1
+                The axis to perform the map across (0 - index, 1 - columns).
+            partitions : NumPy array
+                The partitions of Modin Frame.
+            map_func : callable
+                The function to apply.
+            lengths : list(int)
+                The list of lengths to shuffle the object
+
+        Returns
+        -------
+        NumPy array
+            An array of new partitions for a Modin Frame.
+        """
+        preprocessed_map_func = cls.preprocess_func(map_func)
+        partitions = (
```

Comment:
There is the function [axis_partition](https://github.com/modin-project/modin/blob/fe0afedd051f9aa721d27a63fcb8c67df32d9768/modin/engines/base/frame/partition_manager.py#L78) to get `column_partitions` or `row_partitions`. You need to merge with master before use this.